### DISCUSSION TITLE
How to extract the covariance matrix of a defined model

### DISCUSSION THREAD

**Discussion by Miorat** [GeoStat-Framework/GSTools#327] [Q&A] [answered]
Hi,

Could anyone assist me with how I can extract the covariance matrix of any generated model? 

Thank you in advance,

**Comment by LSchueler** [comment]
Hi,

what exactly do you mean by covariance matrix? - The Gaussian fields used here are completely defined by their [covariance model](https://geostat-framework.readthedocs.io/projects/gstools/en/stable/examples/02_cov_model/index.html). The fields are statistically homogeneous, except for an anisotropy factor along the main axes.

**Comment by MuellerSeb** [comment, marked as answer]
Hey there,

my assumption is, you mean the covariance matrix used in simple-kriging (or any other kriging routine).

I really though it was implemented there, but it seems, that this attribute is missing in the kriging classes.

A temporary workaround to calculate `cov_mat` could look like this:
```python
import gstools as gs
from scipy.spatial.distance import pdist, squareform

# 2D x-y pos pairs for unstructured data
pos = ([2,3,4], [1,3,5])
model = gs.Gaussian(dim=2, nugget=0.5)
iso_pos = model.isometrize(pos)

cov_mat = model.cov_nugget(squareform(pdist(iso_pos.T)))
print(cov_mat)
```

Maybe this helps in the meantime.

Cheers, Sebastian
